### DISCUSSION TITLE
model load error

### DISCUSSION THREAD

**Discussion by brunorbqi** [opencobra/cobrapy#1370] [General]
I was testing the S. aureus rf122 GEM published in the work,doi: [10.1073/pnas.1523199113](https://doi.org/10.1073%2Fpnas.1523199113) , named SA_RF122_uid57661_273036_3.xml. The cobrapy failed in load the model returning the following error:

CobraSBMLError: Something went wrong reading the SBML model. Most likely the SBML model is not valid.

I tried to validate the SBML file using cobrapy and that is what it returns: 

A = cobra.io.sbml.validate_sbml_model('SA_RF122_uid57661_273036_3.xml')
A

(None,
 {'SBML_FATAL': [],
  'SBML_ERROR': ['E0 (Error): XML content (core, L1); Missing XML declaration; Missing XML declaration at beginning of XML input.\n',
   'E1 (Error): General SBML conformance (core, L3); No model definition found; An SBML document must contain a <model> element. The <model> element is optional in L3V2 and beyond.\nReference: L3V1 Section 4.1\n'],
  'SBML_SCHEMA_ERROR': [],
  'SBML_WARNING': [],
  'COBRA_FATAL': [],
  'COBRA_ERROR': ['No SBML model detected in file.'],
  'COBRA_WARNING': [],
  'COBRA_CHECK': []})

I wish to know if there is any way to convert a sbml lvl3 fbc 1 to sbml lvl3 fbc2. Because I suppose that is the issue of this model. Best regards.

**Comment by Midnighter** [comment]
I can't look at the model myself right now, but the error that stands out to me is:

> Missing XML declaration; Missing XML declaration at beginning of XML input.

**Comment by cdiener** [comment]
Checked the ones from the publication and unfortunately those are not valid SBML models. If you add `<?xml version="1.0" encoding="UTF-8"?>` as the first line of the file you will get a bit further but there are still some invalid IDs.
